tests/phpunit/integration/Core/Authentication/Clients/OAuth_ClientTest.php

```diff
@@ -644,20 +645,54 @@ function ( Request $request ) {
 
 		// Verify the redirect URL was stored in transients.
 		$stored_redirect = $transients->get( $code_hash );
-		$this->assertNotEmpty( $stored_redirect, 'Redirect URL was not stored in transients' );
-		$this->assertEquals( $redirect_url, $stored_redirect );
+		$this->assertEquals( $redirect_url, $stored_redirect, 'The redirect URL should be stored in transients with the correct hash' );
 
-		// Verify the key behavior: that the transient is stored with the correct hash and has the right value.
-		$this->assertEquals(
-			$redirect_url,
-			$transients->get( $code_hash ),
-			'The redirect URL should be stored in transients with the correct hash'
+		// SECOND AUTHORIZATION ATTEMPT.
+
+		// Create a new client instance to test the second attempt.
+		$client2 = new OAuth_Client( $context, null, $user_options, null, null, null, null, $transients );
+		$client2->get_authentication_url( $success_redirect );
+
+		// Intercept the redirect to verify the URL is the same as the first attempt.
+		add_filter(
+			'wp_redirect',
+			function ( $location ) use ( $stored_redirect ) {
+				// When `wp_safe_redirect` is called, check the location matches our transient.
+				$this->assertEquals( $stored_redirect, $location, 'Second call should redirect to stored URL' );
+
+				// Throw a RedirectException to simulate the behavior of `wp_redirect`.
+				throw new RedirectException( $location );
+			},
+			10,
+			1
 		);
```

Comment:
Hmm, we aren't adding a `wp_redirect` hook to verify the other redirects in these tests, so I don't these we need to here, this can be removed to simplify the test:

```suggestion
```

Reply:
Removed.